lib/auth/userloginstate/generator.go

```diff
@@ -257,9 +258,12 @@ func (g *Generator) handleAccessListMembership(ctx context.Context, user types.U
 	membershipKind, err := accesslists.IsAccessListMember(ctx, user, accessList, g.accessLists, g.accessLists, g.clock)
 	// Return early if there was an error or the user isn't a member of the access list.
 	if err != nil || membershipKind == accesslistv1.AccessListUserAssignmentType_ACCESS_LIST_USER_ASSIGNMENT_TYPE_UNSPECIFIED {
-		// Log any error.
+		// Log any error besides user being locked.
 		if err != nil {
-			g.log.WarnContext(ctx, "checking access list membership", "error", err)
+			lockedError := trace.IsAccessDenied(err) && strings.Contains(err.Error(), "locked")
```

Comment:
Can we use error wrapping instead and check with `errors.Is()` or `errors.As()`? It was introduced to Go to avoid these sorts of error message string comparisons.

Reply:
The problem here is that the error message is dynamic:

```go
return accesslistv1.AccessListUserAssignmentType_ACCESS_LIST_USER_ASSIGNMENT_TYPE_UNSPECIFIED, trace.AccessDenied("User '%s' is currently locked", user.GetName())
```

The only reasonable way I can think of to make it `trace.AccessDenied` is to do var like:

```go
var ErrUserLocked = trace.AccessDenied("User is currently locked")
```

But then we lose the information but the user. From the looks for the error message format, it looks like it cloud be displayed somewhere. So I'm not sure how to go about it.

Does that look good?

```diff
- trace.AccessDenied("User '%s' is currently locked", user.GetName())
+ trace.Wrap(ErrUserLocked, "user=%q", user.GetName())
```

Reply:
PTALA